exporter/splunkhecexporter/client.go

```diff
@@ -444,7 +444,10 @@ func (c *client) pushMetricsDataInBatches(ctx context.Context, md pmetric.Metric
 // td traces are parsed to Splunk events.
 func (c *client) pushTracesDataInBatches(ctx context.Context, td ptrace.Traces, headers map[string]string) error {
 	buf := c.bufferPool.get()
-	defer c.bufferPool.put(buf)
+	buf.OnClose(func() error {
```

Comment:
you added it only for Traces and not metrics and logs

Reply:
Oops. I will add it to all signals and rerun the benchmarks.

Reply:
unfortunately, new results show that the `Close` method is not called.